client/components/theme/index.jsx

```diff
@@ -62,6 +61,10 @@ var Theme = React.createClass( {
 		actionLabel: React.PropTypes.string
 	},
 
+	shouldComponentUpdate: function( nextProps ) {
+		return ! isEqual( nextProps.theme, this.props.theme );
```

Comment:
To prevent headaches later, it's better to compare the whole theme object — it shouldn't be horrendously slow.